File: Java/SpringBootPrototype/src/main/java/com/hydron/sample/springbootprototype/pojo/Bar.java
package com.hydron.sample.springbootprototype.pojo;

import lombok.*;

import java.time.Instant;

@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@ToString
public class Bar implements ExampleData {
    private String name;
    private String identifier;
    private Instant timestamp;
    private static final String text = "%s [%s] - Bar says \"Goodbye, %s!\"";

    @Override
    public String getMessage() {
        return String.format(text, this.identifier, this.timestamp, this.name);
    }
}

File: Rust/prototype/src/structs/bar.rs
use chrono::{DateTime, Utc};

use crate::structs::example_data::ExampleData;

pub struct Bar {
    pub name: String,
    pub identifier: String,
    pub timestamp: DateTime<Utc>,
}

impl ExampleData for Bar {
    fn get_message(&self) -> String {
        return format!("{} [{}] - Bar says \"Goodbye, {}!\"", self.identifier, self.timestamp, self.name);
    }
}

impl Bar {
    pub fn new(name: &str, identifier: &str, timestamp: DateTime<Utc>) -> Self {
        return Self {
            name: String::from(name),
            identifier: String::from(identifier),
            timestamp,
        };
    }
}
